src/MongoDB.Driver.Core/Core/NativeLibraryLoader/Linux/NativeMethods.cs

```diff
@@ -0,0 +1,61 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using MongoDB.Driver.Core.Misc;
+
+namespace MongoDB.Driver.Core.NativeLibraryLoader.Linux
```

Comment:
This class and file should be called LinuxLibraryLoader.

The P/Invoke declarations can be put inside a nested `NativeMethods` class to suppress the warning you were getting.

There is no need to have multiple classes all with the same name.

See my prototype branch.